packages/backend-function/src/factory.ts

```diff
@@ -209,14 +224,58 @@ class AmplifyFunction
   readonly resources: FunctionResources;
   constructor(scope: Construct, id: string, props: HydratedFunctionProps) {
     super(scope, id);
+    const envVars = {
+      ...props.environment,
+      SECRET_PATH_ENV_VARS: process.env.SECRET_PATH_ENV_VARS,
+    };
+    let bannerCode;
+    const hasSecrets = secretPaths.length > 0;
+
+    if (hasSecrets) {
+      const ext = import.meta.url.split('.').pop();
+      const bannerCodeFile = fileURLToPath(
+        new URL(
+          `./resolve_secret_banner.${ext === 'ts' ? 'ts' : 'js'}`,
```

Comment:
We can use [createRequire](https://nodejs.org/api/module.html#modulecreaterequirefilename) to get the path rather than doing this construction ourself

Reply:
We may also be able to use `require.resolve` but I'm not sure if that plays nice in ESM modules